clients/src/main/java/org/apache/kafka/clients/consumer/internals/IdempotentCloser.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.consumer.internals;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Supplier;
+
+/**
+ * {@code IdempotentCloser} encapsulates some basic logic to ensure that a given resource is only closed once.
+ * It provides methods to invoke callbacks (via optional {@link Runnable}s) for either the <em>initial</em> close
+ * and/or any <em>subsequent</em> closes.
+ *
+ * <p/>
+ *
+ * Here's an example:
+ *
+ * <pre>
+ *
+ * public class MyDataFile implements Closeable {
+ *
+ *     private final IdempotentCloser closer = new IdempotentCloser();
+ *
+ *     private final File file;
+ *
+ *     . . .
+ *
+ *     public boolean write() {
+ *         closer.maybeThrowIllegalStateException(() -> String.format("Data file %s already closed!", file));
+ *         writeToFile();
+ *     }
+ *
+ *
+ *     public boolean isClosed() {
+ *         return closer.isClosed();
+ *     }
+ *
+ *     &#064;Override
+ *     public void close() {
+ *         Runnable onInitialClose = () -> {
+ *             cleanUpFile(file);
+ *             log.debug("Data file {} closed", file);
+ *         };
+ *         Runnable onSubsequentClose = () -> {
+ *             log.warn("Data file {} already closed!", file);
+ *         };
+ *         closer.close(onInitialClose, onSubsequentClose);
+ *     }
+ * }
+ * </pre>
+ *
+ * Note that the callbacks are optional and if unused operates as a simple means to ensure resources
+ * are only closed once.
+ */
+public class IdempotentCloser implements AutoCloseable {
+
+    private final AtomicBoolean flag;
```

Comment:
This flag actually means that the closer isClosed right? Wouldn't be more straightforward to name it like that?

Reply:
Yes. Changed to `isClosed` as suggested.